sharding/txpool/service.go

```diff
@@ -2,27 +2,60 @@
 package txpool
 
 import (
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/sharding"
 )
 
 // ShardTXPool handles a transaction pool for a sharded system.
 type ShardTXPool struct {
-	p2p sharding.ShardP2P
+	p2p              sharding.ShardP2P
```

Comment:
We decided on not using interfaces for things we most likely won't have a second implementation for. See #177 - just use *p2p.Server.